src/rudderstack.ts

```diff
@@ -81,6 +81,34 @@ export class RudderStack {
         }
     }
 
+    //tracking the track js error
+
+    trackConsoleErrors = (): any => {
+        let latestErrorMessage: string | null = null
+        // Preserve the original console.error function
+        const originalConsoleError = console.error
+
+        // Override console.error to capture and handle errors
+        console.error = function (...args: unknown[]): any {
+            // Log the error to the console as usual
+            originalConsoleError.apply(console, args)
+
+            // Create a clean error message without __trackjs_state__
+            const errorMessage = args
+                .map(arg =>
+                    arg && typeof arg === 'object' && 'message' in arg
+                        ? (arg as Error).message
+                        : typeof arg === 'object'
+                        ? JSON.stringify(arg, (key, value) => (key.startsWith('__trackjs') ? undefined : value))
+                        : String(arg)
+                )
+                .join(' ')
+            latestErrorMessage = errorMessage
+            return errorMessage
+        }
+        return latestErrorMessage
+    }
```

Comment:
@Aswathy Menon we need to add this in `cacheTrackEvents` instead if we want to capture all errors. Also this function won't work you are returning the message but it will always be empty, it will only have value when error occurs but when that happened you won't get the hold of the value anymore.